packages/core/src/entity/ArrayCollection.ts

```diff
@@ -345,6 +349,10 @@ export class ArrayCollection<T extends object, O extends object> {
     return this.count() === 0;
   }
 
+  last(): T | undefined {
+    return this.getItems()[this.items.size - 1];
```

Comment:
same here, you can just check the loaded state and return `this[this.length - 1]`. in fact `this.length` is a getter that will check the loaded state, so here it should be enough to just do this
```suggestion
    return this[this.length - 1];
```

Reply:
i guess you meant
```ts

  first(): T | undefined {
    return this[0];
  }

```